scrape/target.go

```diff
@@ -347,6 +347,18 @@ func (app *limitAppender) Append(ref storage.SeriesRef, lset labels.Labels, t in
 	return ref, nil
 }
 
+func (app *limitAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (storage.SeriesRef, error) {
+	app.i++
```

Comment:
We need the staleness check here, too. There are histogram staleness markers where the sum is a StaleNaN.